mesa/time.py

```diff
@@ -74,20 +81,28 @@ def add(self, agent: Agent) -> None:
             agent: An Agent to be added to the schedule. NOTE: The agent must
             have a step() method.
         """
-        if agent.unique_id in self._agents:
-            raise Exception(
-                f"Agent with unique id {agent.unique_id!r} already added to scheduler"
-            )
 
-        self._agents[agent.unique_id] = agent
+        if agent not in self._agents:
+            self._agents.add(agent)
+        else:
+            raise ValueError("agent already added to scheduler")
 
     def remove(self, agent: Agent) -> None:
         """Remove all instances of a given agent from the schedule.
 
         Args:
             agent: An agent object.
         """
-        del self._agents[agent.unique_id]
+        if not self._remove_warning_given:
+            self._remove_warning_given = True
+        warnings.warn(
+            "Because of the shift to using weakrefs, it is no longer needed to explicitly remove"
```

Comment:
I think this holds up as long a you use a single scheduler, and assume agents in model = agents in schedulers. However, as you let that assumption go you might want to remove agents from a scheduler but not from a model.

Reply:
However, I find it weird you still have to explicitly add agents to the scheduler, but don't explicitly have to remove them.

Reply:
This comes back to having one or more than one scheduler. If you have only one, then yes, there is no need to remove them. But that would be a backward incompatible change and thus is outside the scope of this PR.